src/ServiceControl/Operations/AuditImporter.cs

```diff
@@ -104,24 +105,27 @@ public async Task<ProcessedMessage> ConvertToSaveMessage(MessageContext message)
                 messageId = DeterministicGuid.MakeId(message.MessageId).ToString();
             }
 
-            var metadata = new Dictionary<string, object>
+            var metadata = new ConcurrentDictionary<string, object>
             {
                 ["MessageId"] = messageId,
                 ["MessageIntent"] = message.Headers.MessageIntent(),
                 ["HeadersForSearching"] = string.Join(" ", message.Headers.Values)
             };
 
-            // TODO: Fan out?
+            var enricherTasks = new List<Task>(enrichers.Length);
+            // ReSharper disable once LoopCanBeConvertedToQuery
             foreach (var enricher in enrichers)
             {
-                await enricher.Enrich(message.Headers, metadata)
-                    .ConfigureAwait(false);
+                enricherTasks.Add(enricher.Enrich(message.Headers, metadata));
             }
 
+            await Task.WhenAll(enricherTasks)
+                .ConfigureAwait(false);
+
             await bodyStorageEnricher.StoreAuditMessageBody(message.Body, message.Headers, metadata)
                 .ConfigureAwait(false);
 
-            var auditMessage = new ProcessedMessage(message.Headers, metadata)
+            var auditMessage = new ProcessedMessage(message.Headers, new Dictionary<string, object>(metadata))
```

Comment:
This would be the consequence but I think this is an OK price to pay given that we can fan out the enrichers which do domain event raise and other IO bound stuff?